src/TestFramework/Factory.php

```diff
@@ -146,20 +146,17 @@ public function create(string $adapterName, bool $skipCoverage): TestFrameworkAd
     /**
      * Get only those source files that will be mutated to use them in coverage whitelist
      *
-     * @return list<string>
+     * @return list<SplFileInfo>
      */
     private function getFilteredSourceFilesToMutate(): array
     {
         if ($this->sourceFileFilter->getFilters() === []) {
             return [];
         }
 
-        /** @var list<string> $filteredPaths */
-        $filteredPaths = array_filter(array_map(
-            static fn (SplFileInfo $file) => $file->getRealPath(),
-            iterator_to_array($this->sourceFileFilter->filter($this->infectionConfig->getSourceFiles()))
-        ));
+        /** @var list<SplFileInfo> $files */
```

Comment:
is that phpdoc really necessary?

Reply:
short answer: yes, because deeper in the stack we have return type `iterable<SplFileInfo|Trace>` and here we need to specify it.

In other word - this type was here, and removing of this `@var` produces a phpstan error.

I tried quickly to replace `iterable<SplFileInfo|Trace>`  with `iterable<T>` in the stack - but that produces too many errors that need to be analyzed separately. Something wrong with types across those classes.